### PR TITLE
PIN-3149 Status 404 su GET /clients subito dopo cancellazione di un client

### PR DESCRIPTION
**Pull request by nttdata-rtorsoli**

### CODE REVIEW

**Review by galales — CHANGES_REQUESTED** [review]
the new authorization-process endpoint should not be invoked as a separated BFF endpoint, but in the existing `/clients` in the BFF

File: src/main/resources/interface-specification.yml [hunk 1]
```diff
@@ -8763,18 +8763,44 @@ components:
...
         - CONSUMER
-    CompactClients:
+    CompactClientsWithKeys:
       type: object
       properties:
         results:
           type: array
           items:
-            $ref: '#/components/schemas/CompactClient'
+            $ref: '#/components/schemas/CompactClientWithKeys'
```
Comment (galales) [inline src/main/resources/interface-specification.yml]: we should keep the previous definition

**Comment by TonioGela** [comment]
Is this PR actual? I do see some CI errors and unanswered discussions

File: src/main/resources/interface-specification.yml [hunk 2]
```diff
@@ -8826,6 +8826,16 @@ components:
...
+    CompactKey:
+      type: object
+      properties:
+        id:
+          type: string
+        name:
+          type: string
+      required:
+        - id
+        - name
```
Comment (galales) [inline src/main/resources/interface-specification.yml]: no more required 
```suggestion
```

File: src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/package.scala [hunk 3]
```diff
@@ -194,6 +192,11 @@ package object impl extends SprayJsonSupport with DefaultJsonProtocol {
   implicit val publicKeysFormat: RootJsonFormat[PublicKeys]     =
     jsonFormat1(PublicKeys)
 
+  implicit val compactKeyFormat: RootJsonFormat[CompactKey]         =
+    jsonFormat2(CompactKey)
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/package.scala]: No more required
```suggestion
```

File: src/main/scala/it/pagopa/interop/backendforfrontend/service/types/AuthorizationProcessServiceTypes.scala [hunk 4]
```diff
@@ -16,6 +16,16 @@ object AuthorizationProcessServiceTypes {
...
 
+  implicit class ClientEntryKeyProcessConverter(private val c: AuthorizationProcess.ClientWithKeys) extends AnyVal {
+    def toApi: CompactClient =
+      CompactClient(id = c.client.id, name = c.client.name, hasKeys = c.keys.size > 0)
+  }
+
+  implicit class KeyEntryProcessConverter(private val k: AuthorizationProcess.KeyEntry) extends AnyVal {
+    def toApi: CompactKey =
+      CompactKey(id = k.id, name = k.name)
+  }
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/backendforfrontend/service/types/AuthorizationProcessServiceTypes.scala]: No more required
```suggestion
```

File: src/main/scala/it/pagopa/interop/backendforfrontend/service/types/AuthorizationProcessServiceTypes.scala [hunk 5]
```diff
@@ -16,6 +16,16 @@ object AuthorizationProcessServiceTypes {
       CompactClient(id = c.id, name = c.name, hasKeys = hasKeys)
   }
 
+  implicit class ClientEntryKeyProcessConverter(private val c: AuthorizationProcess.ClientWithKeys) extends AnyVal {
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/backendforfrontend/service/types/AuthorizationProcessServiceTypes.scala]: can you please move it back to minimize code changes?

File: src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/package.scala [hunk 6]
```diff
@@ -194,6 +192,11 @@ package object impl extends SprayJsonSupport with DefaultJsonProtocol {
...
+  implicit val compactClientsFormat: RootJsonFormat[CompactClients] =
+    jsonFormat2(CompactClients)
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/package.scala]: can you please move it back to minimize code changes?

File: src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/ClientsApiMarshallerImpl.scala [hunk 7]
```diff
@@ -40,4 +37,8 @@ object ClientsApiMarshallerImpl extends ClientsApiMarshaller with SprayJsonSuppo
   override implicit def toEntityMarshallerClient: ToEntityMarshaller[Client] = sprayJsonMarshaller[Client]
 
   override implicit def toEntityMarshallerOperator: ToEntityMarshaller[Operator] = sprayJsonMarshaller[Operator]
+
+  override implicit def toEntityMarshallerCompactClients: ToEntityMarshaller[CompactClients] =
+    sprayJsonMarshaller[CompactClients]
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/ClientsApiMarshallerImpl.scala]: can you please move it back to minimize code changes?

**Review by TonioGela — APPROVED** [review]

File: src/main/scala/it/pagopa/interop/backendforfrontend/service/types/AuthorizationProcessServiceTypes.scala (hunk 4 above)
Comment (nttdata-rtorsoli) [reply src/main/scala/it/pagopa/interop/backendforfrontend/service/types/AuthorizationProcessServiceTypes.scala]: implicit class CompactClientConverter(private val c: AuthorizationProcess.Client) was a duplicate

**Review by galales — APPROVED** [review]

**Pull request merged**